src/pip/_internal/resolution/legacy/resolver.py

```diff
@@ -256,6 +260,34 @@ def _check_skip_installed(self, req_to_install):
         self._set_req_to_reinstall(req_to_install)
         return None
 
+    def _populate_link(self, req):
```

Comment:
Ooh, nice.

My brain is too fixed at thinking of populate_link as a method of InstallRequirement to have thought of doing this.